pkg/controller/ephemeralcluster/reconciler.go

```diff
@@ -118,10 +131,14 @@ func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	return reconcile.Result{}, nil
 }
 
-func (r *reconciler) handleGetProwJobError(err error) (reconcile.Result, error) {
+func (r *reconciler) handleGetProwJobError(ctx context.Context, ec *ephemeralclusterv1.EphemeralCluster, err error) (reconcile.Result, error) {
 	if kerrors.IsNotFound(err) {
-		// TODO
-		return reconcile.Result{}, reconcile.TerminalError(errors.New("unimplemented"))
+		finalizers, removed := removeElement(ec.Finalizers, DependentProwJobFinalizer)
```

Comment:
probably `DependentProwJobFinalizer` defined in this file?

Reply:
Yes, correct. I'll address it in an upcoming PR as I already have an LGTM on this.

Reply:
Nevermind, I had to push something else anyway.